src/py/pyodide/package_loader.py

```diff
@@ -0,0 +1,39 @@
+import tarfile
```

Comment:
Functions need documentation type annotation, and maybe let's make this `_package_loader` if we don't intend to make it public?